src/libaktualizr/primary/sotauptaneclient.cc

```diff
@@ -16,23 +16,53 @@
 #include "uptane/secondaryfactory.h"
 #include "utilities/utils.h"
 
-SotaUptaneClient::SotaUptaneClient(Config &config_in, std::shared_ptr<event::Channel> events_channel_in,
-                                   Uptane::Manifest &uptane_manifest_in, std::shared_ptr<INvStorage> storage_in,
-                                   HttpInterface &http_client, const Bootloader &bootloader_in,
-                                   ReportQueue &report_queue_in)
+std::shared_ptr<SotaUptaneClient> SotaUptaneClient::newDefaultClient(
+    Config &config_in, std::shared_ptr<INvStorage> storage_in, std::shared_ptr<event::Channel> events_channel_in) {
+  std::shared_ptr<HttpClient> http_client_in = std::make_shared<HttpClient>();
+  std::shared_ptr<Uptane::Fetcher> uptane_fetcher =
+      std::make_shared<Uptane::Fetcher>(config_in, storage_in, http_client_in);
+  std::shared_ptr<Bootloader> bootloader_in = std::make_shared<Bootloader>(config_in.bootloader);
+  std::shared_ptr<ReportQueue> report_queue_in = std::make_shared<ReportQueue>(config_in, http_client_in);
+
+  return std::make_shared<SotaUptaneClient>(config_in, storage_in, http_client_in, uptane_fetcher, bootloader_in,
+                                            report_queue_in, events_channel_in);
+}
+
+std::shared_ptr<SotaUptaneClient> SotaUptaneClient::newTestClient(Config &config_in,
+                                                                  std::shared_ptr<INvStorage> storage_in,
+                                                                  std::shared_ptr<HttpInterface> http_client_in,
+                                                                  std::shared_ptr<event::Channel> events_channel_in) {
+  std::shared_ptr<Uptane::Fetcher> uptane_fetcher =
+      std::make_shared<Uptane::Fetcher>(config_in, storage_in, http_client_in);
+  std::shared_ptr<Bootloader> bootloader_in = std::make_shared<Bootloader>(config_in.bootloader);
```

Comment:
Do we ever use a nonstandard bootloader?

Reply:
Currently no